test/hotspot/jtreg/compiler/loopopts/superword/TestSubwordVectorization.java

```diff
@@ -0,0 +1,155 @@
+/*
+ * Copyright (c) 2025, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+package compiler.loopopts.superword;
+
+import compiler.lib.ir_framework.*;
+import java.util.Random;
+import jdk.test.lib.Utils;
+
+/*
+ * @test
+ * @bug 8342095
+ * @key randomness
+ * @summary Ensure that vectorization of conversions between subword types works as expected.
+ * @library /test/lib /
+ * @run driver compiler.loopopts.superword.TestSubwordVectorization
+ */
+
+public class TestSubwordVectorization {
+    private static final Random RANDOM = Utils.getRandomInstance();
+    private static final int SIZE = 1024;
+
+    public static void main(String[] args) {
+        TestFramework.run();
+    }
+
+    @Setup
+    static Object[] setupIntArray() {
+        int[] res = new int[SIZE];
+
+        for (int i = 0; i < SIZE; i++) {
+            res[i] = RANDOM.nextInt();
+        }
```

Comment:
Can you please use `Generators.java`? It would be great if we can use that in the future, to create more "interesting" input data ;)